File: nodejs/domestic/02_create_domestic_transfer.js
const { sendEasylinkRequest } = require('../helpers');

const baseUrl       = 'https://sandbox.easylink.id';
const appKey        = 'YOUR_APP_KEY';
const privateKeyPem = '/path/to/private.pem';
const accessToken   = 'YOUR_ACCESS_TOKEN';

(async () => {
  const payload = {
    partnerReferenceNo: `REF-${Date.now()}`,
    amount: 100000,
    bankCode: 'BCA',
    accountNumber: '1234567890',
    recipientName: 'John Doe',
    remark: 'Payment for Order #1001',
  };

  console.log('Creating Domestic Transfer...');
  try {
    const res = await sendEasylinkRequest(
      baseUrl,
      '/v2/transfer/create-domestic-transfer',
      'POST',
      payload,
      appKey,
      privateKeyPem,
      accessToken
    );
    console.log('Response:\n', JSON.stringify(res.data, null, 2));
  } catch (err) {
    console.error('Error:', err.message);
  }
})();

File: python/domestic/02_create_domestic_transfer.py
import sys
import time
import json
from pathlib import Path
sys.path.append(str(Path(__file__).resolve().parent.parent))

from helpers import send_easylink_request

base_url        = 'https://sandbox.easylink.id'
app_key         = 'YOUR_APP_KEY'
private_key_pem = '/path/to/private.pem'
access_token    = 'YOUR_ACCESS_TOKEN'

payload = {
    'partnerReferenceNo': f"REF-{int(time.time())}",
    'amount': 100000,
    'bankCode': 'BCA',
    'accountNumber': '1234567890',
    'recipientName': 'John Doe',
    'remark': 'Payment for Order #1001'
}

print("Creating Domestic Transfer...")
try:
    res = send_easylink_request(
        base_url,
        '/v2/transfer/create-domestic-transfer',
        'POST',
        payload,
        app_key,
        private_key_pem,
        access_token
    )
    print("Response:\n", json.dumps(res['data'], indent=2))
except Exception as e:
    print("Error:", e)
